File: 1966-frequency-of-the-most-frequent-element/1966-frequency-of-the-most-frequent-element.java
class Solution {
    public int maxFrequency(int[] nums, int k) {
        int l = 0;
        int max = 0;
        long sum = 0;
        Arrays.sort(nums);

        for (int r = 0; r < nums.length; r++) {
            sum += nums[r];

            while ((long)nums[r] * (r - l + 1) - sum > k) {
                sum -= nums[l];
                l++;
            }
            max = Math.max(max, r - l + 1);
        }
        return max;
    }
}

File: 1966-frequency-of-the-most-frequent-element/1966-frequency-of-the-most-frequent-element.cpp
class Solution {
public:
    int maxFrequency(vector<int>& nums, int k) {
        int l = 0;
        int r  = 0;
        int n = nums.size();
        sort(nums.begin(),nums.end());

        long long currSum = 0;
        int maxLen = 1;
        while(r<n){
            currSum += nums[r];
            while(currSum+k <   (nums[r]*1LL*(r-l+1))){
                currSum-=nums[l];
                l++;
            }
            maxLen = max(maxLen,r-l+1);
            r++;
        }   
        return maxLen;
    }
};